### ISSUE TITLE
non-cloud tests on travis should be run without gcloud logged in

### ISSUE THREAD

**Issue by lbergelson** [broadinstitute/gatk#2706]
We recently discovered that some of the tests we didn't think required google cloud authentication require that gcloud be initialized.

Travis didn't catch this because we always initialize gcloud in order to do log uploading.  We should change this so it's only initialized during the tests for the cloud tests.  

The actual error we discovered didn't require that credentials be correct, only that a default project had been configured so simply logging out isn't enough to trigger it.

**Comment by droazen** [comment]
Closing -- I looked at these tests, and it's my belief that all of them should be marked as "bucket", except for one test in `PSUtils` that was poorly written and fails when run as root.

**Comment by droazen** [comment]
Re-opening, as I misunderstood the purpose of this ticket :)

**Comment by lbergelson** [comment]
@droazen I actually don't think most of those tests should be `cloud` tests. 
 BucketUtilsTest.testIsCloudStorageURL for instance. It seems dangerous if that method requires being logged into gcloud.  Separate issue for that here: https://github.com/broadinstitute/gatk/issues/2707

**Comment by lbergelson** [comment]
calling `isCloudStorageUrl` should be decidable without logging in.

**Comment by droazen** [comment]
Yes, we can fix that separately (probably will be fixed as a side effect of introducing a URL class into GATK). But the tests that Lee is talking about should be marked as bucket, since they are doing more than that.

**Comment by lbergelson** [comment]
| Test failure | Fix |
 | ------| ------ |
 |  FeatureInputUnitTest. testGcsPathAndName[0](gs://bucket/user/my.vcf, gs://bucket/user/my.vcf, gs://bucket/user/my.vcf) |  seems like this should be doable unauthenticated   |
 | PSUtilsTest. testWriteTwoKryo | #2708 |
 | BucketUtilsTest.testGetPathOnGcsDirectory | it seems like this should be doable unauthenticated #2707 |
 | BucketUtilsTest. testIsCloudStorageURL | #2707  |
 | GcsNioIntegrationTest. openPublicFile | mark as bucket |
 | GcsNioIntegrationTest. testGcsEnabled | mark as bucket |

**Comment by droazen** [comment]
We're marking all of these as bucket for now @lbergelson to unblock our work (with the exception of the `PSUtilsUnitTest`). They require authentication now, so need to be properly marked. The underlying issue should ultimately be fixed by https://github.com/broadinstitute/gatk/issues/958

**Comment by lbergelson** [comment]
@droazen 👍 Sounds good.
